tests/hooks/test_postgres_hook.py

```diff
@@ -24,6 +24,68 @@
 from tempfile import NamedTemporaryFile
 
 from airflow.hooks.postgres_hook import PostgresHook
+from airflow.models import Connection
+
+
+class TestPostgresHookConn(unittest.TestCase):
+
+    def setUp(self):
+        super().setUp()
+
+        self.connection = Connection(
+            login='login',
+            password='password',
+            host='host',
+            schema='schema'
+        )
+
+        self.db_hook = PostgresHook()
+        self.db_hook.get_connection = mock.Mock()
+        self.db_hook.get_connection.return_value = self.connection
+
+    @mock.patch('airflow.hooks.postgres_hook.psycopg2.connect')
+    def test_get_conn(self, mock_connect):
+        self.db_hook.get_conn()
+        assert mock_connect.call_count == 1
+        args, kwargs = mock_connect.call_args
+        self.assertEqual(args, ())
+        self.assertEqual(kwargs['user'], 'login')
+        self.assertEqual(kwargs['password'], 'password')
+        self.assertEqual(kwargs['host'], 'host')
+        self.assertEqual(kwargs['dbname'], 'schema')
+
+    @mock.patch('airflow.hooks.postgres_hook.psycopg2.connect')
+    @mock.patch('airflow.contrib.hooks.aws_hook.AwsHook.get_client_type')
+    def test_get_conn_rds_iam_postgres(self, mock_client, mock_connect):
+        self.connection.extra = '{"iam":true}'
+        mock_client.return_value.generate_db_auth_token.return_value = 'aws_token'
+        self.db_hook.get_conn()
+        assert mock_connect.call_count == 1
+        assert mock_client.call_count == 1
+        args, kwargs = mock_connect.call_args
+        self.assertEqual(args, ())
+        self.assertEqual(kwargs['user'], 'login')
+        self.assertEqual(kwargs['password'], 'aws_token')
+        self.assertEqual(kwargs['host'], 'host')
+        self.assertEqual(kwargs['dbname'], 'schema')
+
+    @mock.patch('airflow.hooks.postgres_hook.psycopg2.connect')
+    @mock.patch('airflow.contrib.hooks.aws_hook.AwsHook.get_client_type')
+    def test_get_conn_rds_iam_redshift(self, mock_client, mock_connect):
+        self.connection.extra = '{"iam":true, "redshift":{"cluster-identifier":"my_cluster_id"}}'
```

Comment:
Random thought: should the host map to the cluster-identifer somehow? Is it possible to go from hostname (in the existing `.host` field) to the cluster identifier?

Reply:
good idea. done